datafusion/common/src/scalar.rs

```diff
@@ -726,6 +726,24 @@ impl ScalarValue {
         }
     }
 
+    /// whether this value is true or not.
+    pub fn is_true(&self) -> bool {
+        match self {
+            ScalarValue::Boolean(v) => v.eq(&Some(true)),
+            ScalarValue::Null => false,
+            other => panic!("Cannot apply 'IS TRUE' to arguments of type '{} IS TRUE'. Supported form(s): '<BOOLEAN> IS TRUE'", other)
```

Comment:
This code is called from `IsTrueExpr.evaluate`, which returns a `Result`. I think we should move this logic out of `ScalarValue` and have it in `IsTrueExpr` so that we can return an `Err` and avoid a `panic!`.

Reply:
Okay, let me know if my changes in `is_true.rs` and `is_false.rs` are what you had in mind.